environment/src/main/java/com/sequenceiq/environment/environment/flow/stop/handler/StopFreeIpaHandler.java

```diff
@@ -33,7 +35,9 @@ public String selector() {
     public void accept(Event<EnvironmentDto> environmentDtoEvent) {
         EnvironmentDto environmentDto = environmentDtoEvent.getData();
         try {
-            freeIpaPollerService.stopAttachedFreeipaInstances(environmentDto.getId(), environmentDto.getResourceCrn());
+            if (Objects.isNull(environmentDto.getParentEnvironmentCrn())) {
```

Comment:
Strings.isNullOrEmtpy?

Reply:
Nice catch, done both here and in the other PR